.github/workflows/openyurt-unit-test.yml

```diff
@@ -0,0 +1,119 @@
+name: Build and Test OpenYurt Deployer
+
+on:
+    push:
+      branches: [ main, legacy-firecracker-v0.24.0-with-upf-support ]
+      paths-ignore:
+      - 'docs/**'
+      - '**.md'
+    pull_request:
+      branches: [ main, legacy-firecracker-v0.24.0-with-upf-support, openyurt ]
+      paths-ignore:
+      - 'docs/**'
+      - '**.md'
+    workflow_dispatch:
+
+env:
+    GOOS: linux
+    GO111MODULE: on
+
+jobs:
+  main-unit-test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Set up Go 1.19
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.19'
+
+    - name: Check out the code
+      uses: actions/checkout@v4
+
+    - name: Build scripts
+      run: 
+        pushd scripts/openyurt-deployer && go build -o oy_deploy && popd
+
+    - name: Allow root login
+      run: |
+        echo "PermitRootLogin=yes" | sudo tee -a /etc/ssh/sshd_config
+
+    - name: Set up SSH and Run Test
```

Comment:
combine all the setup steps in a single step where you execute a single setup script. Run the actual test(s) in a separate step